### DISCUSSION TITLE
Leveraging the to_json from_json() with nlohmann::json*

### DISCUSSION THREAD

**Discussion by mynameismichael** [nlohmann/json#2611] [Q&A]
Hi,

I am using this json library and I have a specific use case. I am parsing URI's for a REST API to modify parameters on the fly (to an embedded device) via a web interface. 

So far, I have added the macros for serialisation and they are a dream. They work great, but the real issue is now I need to create a hash table for efficiency to map URI’s to types that can be serialised/deserialised. 

The issue I am encountering is that I would like to do something as follows:

```cpp
// Assume this struct works fine, it’s just an example
struct Person
{
    std::string name;
    Uint8_t age;
    std::string gender;
};
NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE( Person, name, age, gender )

// Example of an entity in a hash-table. 
struct Example
{
    char* name; 
    nlohmann::json* ptr; 
};

Person person;
person.name = "Me";
person.age = 200;
person.gender = "Whatever";

Example entity;
entity.name = "ME";
entity.ptr = (nlohmann::json*)&person; // This doesn't work, of course. Do I need a custom constructor or overloaded operator?

```
I am able to benefit from the =operator auto serialisation of my data structures with the macro, but also I need the nlohmann::json object to be a pointer, because in operation, the data will be changing and I cannot have my hash-table storing old data. Also, to regularly refresh the hash table would be expensive, as there would be many items.

So whilst my data structures all vary in types (many custom data structures), they have benefitted from the macros. But now I need some kind of reference or pointer to them all with the same type, rather than lots of functions, I was hoping this could be achieved with nlohmann::json* but it doesn’t appear to be the case.

I am wondering if anybody has ever tried such a thing, or an area of the documentation you may point me to. I feel what I would like to achieve shouldn’t be so difficult. 

Am I missing something?

**Comment by nlohmann** [comment]
If you want to store a pointer to a JSON value built out of a `Person` struct, where would the JSON value be stored? That is, to what memory should that pointer point?

**Reply by mynameismichael** [reply]
I see your point.

Can you see any way of solving this problem with your library, or any documentation I might look further in to? Happy to extend the library for small amendments.

The ideal situation was to have a same type for everything in the hash-table. I cannot see any other way. Templating doesn't appear to resolve this issue, of course.

**Reply by nlohmann** [reply]
I wouldn't know how to do this.

**Reply by mynameismichael** [reply]
No problem, Niels. I will think on this some more and if I find a solution, I will return to update this discussion. Appreciate the response.
